### DISCUSSION TITLE
Report with Parameter and Default Value

### DISCUSSION THREAD

**Discussion by EAwesomeK** [eclipse-birt/birt#1213] [General]
Is it possible to create a report that requires a value but if you dont provide a value it behaves in a default way?  For example, I need to select to return some data for a specific item so I create a parameter that prompts for a report but if I DON'T provide a value it returns ALL of the items?

**Comment by speckyspooky** [comment]
The easiest way which I use in my reports for "list box" is that I define the the parameter like required
and on the dynamic data set for the list box I add a special key & label for "<all entries>" and with the according sorting this entry will be the first record.
In this way it will be selected by default from BIRT.

Simple example like SQL-style:

SELECT -1 AS key, '<all entries>' AS label, -1 AS sort_order
UNION ALL
SELECT item_id AS key, item_label AS label, <some number> AS sort_order

In this way you get also the user the information that all items will be selected if no item is selected.
This will work with single-selection (, not so good with multi selection).

Another way would be if the parameter is "optional" then you cann check the according value
on your dataset on the parameter register-page. There you cann check the parameter value and if the value is not set/is null then you can set the value to an ALL value which you can use in your query-condition to select all items instead of one special item.

But in all it is your task to prepare the parameter values with the according value options.
